server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolLexer.g4

```diff
@@ -15,9 +15,59 @@
 lexer grammar CobolLexer;
 import CICSLexer;
 channels{TECHNICAL}
+@header {
+  import java.util.regex.Matcher;
+  import java.util.regex.Pattern;
+  import org.apache.commons.lang3.StringUtils;
+}
+@lexer::members {
+private Pattern titlePattern = Pattern.compile("TITLE\\s*(.*?)\\.?\\s\\n.*", Pattern.CASE_INSENSITIVE);
+    private void isTitlePresent() {
+    String input = _input.getText(Interval.of(_tokenStartCharIndex, _input.index()));
+          if(input!=null) {
+                Matcher matcher = titlePattern.matcher(input);
+                if(matcher.matches()){
+                String matchedTitle = matcher.group(1);
+                if(matchedTitle.isEmpty())
+                reportWrongArguments("lexer.titleCompilerDirective");
+                }
+          }
+        }
+    private void reportWrongArguments(String msg) {
+        ANTLRErrorListener listener = getErrorListenerDispatch();
+        String text = _input.getText(Interval.of(_tokenStartCharIndex, _input.index()));
+	    int stop = _input.index();
+	    if(text.contains("\r")) {
```

Comment:
What about \n, \r\n?

Reply:
That won't be a problem, we just need a subset of token.

Reply:
The initial inclination was to get the exact token and have our own custom tokens (extends CommonToken).  But changed it coz it was difficult to get the token.